File: Problem25.py

def fib(limit):
    a, b = 1, 0
    index = 0
    while len(str(a)) < limit:
        a, b = a+b, a
        index += 1
    else:
        return (index+1, len(str(a)))

if __name__ == '__main__':
    answer = fib(20000)
    print(answer[0])

File: Problem25.go
package main

import "fmt"

func fibLimit(limit int) int{

  var a, b int64 = 1, 0
  index := 0

  for IntLen(a) <= limit{
    temp := a
    a = a + b
    b = temp
    index += 1
  }

  return index
}

func IntLen(num int64) int{

  length := 0;

  for num >= 1 {
    num /= 10
    length += 1
  }

  return length
}

func main(){

  fmt.Println(fibLimit(1000))

}
